app/main.py

```diff
@@ -1,3 +1,16 @@
+sorted_variables = {
+    "mutable": [],
+    "immutable": [],
+}
```

Comment:
The `sorted_variables` dictionary is defined outside the `categorize_variables` function. This can lead to issues if the function is called multiple times, as the dictionary will retain values from previous calls. Consider defining it inside the function or resetting it at the beginning of the function.